port/unix/omrsysinfo.c

```diff
@@ -604,20 +608,77 @@ omrsysinfo_get_OS_version(struct OMRPortLibrary *portLibrary)
 {
 	if (NULL == PPG_si_osVersion) {
 		int rc;
-#if !defined (OSX)
+#if defined (OSX)
+		char *versionString = NULL;
+		char *allocatedFileBuffer = NULL;
+#else
 		struct utsname sysinfo;
 #endif
 
 #ifdef J9ZOS390
 		rc = __osname(&sysinfo);
 #elif defined (OSX)
-		/* uname() on MacOS returns the version of the Darwin kernel.
+		/* uname() on MacOS returns the version of the Darwin kernel, not the operating system product name.
 		 * For compatibility with the reference implementation,
 		 * we need the operating system product name.
-		 * This is provided by sysctlbyname(), which is deprecated on Linux.
+		 * On newer versions of MacOS, the OS name is provided by sysctlbyname(), which is deprecated on Linux.
```

Comment:
I couldn't find any connection between `sysctlbyname()` and Linux. Perhaps you meant BSD?

Reply:
I added the comment to clarify why I didn't use the same call on Linux.

The linux `man 2 sysctl` page says:

>  use of this system call  has
>        long  been  discouraged,  and it is so unloved that it is likely to disappear in a future kernel version.  Since Linux 2.6.24, uses of this
>        system call result in warnings in the kernel log.

sysctlbyname doesn't even exist on Linux.

Reply:
I still think the reference to Linux is distracting and inappropriate for this OSX-specific code block.